test/logster/test_defer_logger.rb

```diff
@@ -0,0 +1,34 @@
+require_relative '../test_helper'
+require 'logster/defer_logger'
+require 'logster/logger'
+
+class TestDeferLogger < Minitest::Test
+  def setup
+    @store = TestStore.new
+    @defer_logger = Logster::DeferLogger.new(@store)
+  end
+
+  def test_defer_logger_inherits_logger
+    assert(Logster::Logger === @defer_logger)
+  end
+
+  def test_work_is_done_async
+    queue = Logster::Scheduler.queue
+    assert_equal(0, queue.size)
+
+    @defer_logger.add(4, "hi this a test", "prog")
+
+    assert_equal(1, queue.size)
+    queue << nil
```

Comment:
This is okish ... but I would go with an explicit special message `queue << :terminate` maybe

Reply:
Makes sense, done!